File: server/router/business/cmsBlog.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type CmsBlogRouter struct {
}

// InitCmsBlogRouter 初始化 CmsBlog 路由信息
func (s *CmsBlogRouter) InitCmsBlogRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsBlog").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsBlog")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsBlogApi
	{
		router.POST("createCmsBlog", apiV1.CreateCmsBlog)   // 新建CmsBlog
		router.POST("deleteCmsBlog", apiV1.DeleteCmsBlog) // 删除CmsBlog
		router.POST("deleteCmsBlogByIds", apiV1.DeleteCmsBlogByIds) // 批量删除CmsBlog
		router.POST("updateCmsBlog", apiV1.UpdateCmsBlog)    // 更新CmsBlog
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findCmsBlog", apiV1.FindCmsBlog)        // 根据ID获取CmsBlog
		routerNoRecord.GET("getCmsBlogList", apiV1.GetCmsBlogList)  // 获取CmsBlog列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel CmsBlog列表
	}
}

File: web/src/api/cmsBlog.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsBlog
// @Summary 创建CmsBlog
export const createCmsBlog = (data?: object) => {
   return http.request<resData>("post", "/cmsBlog/createCmsBlog", { data });  
}

// @Tags CmsBlog
// @Summary 删除CmsBlog
export const deleteCmsBlog = (data?: object) => {
   return http.request<resData>("post", "/cmsBlog/deleteCmsBlog", { data });  
}
 

// @Tags CmsBlog
// @Summary 删除ByIdsCmsBlog
export const deleteCmsBlogByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsBlog/deleteCmsBlogByIds", { data });  
}
 

// @Tags CmsBlog
// @Summary 更新CmsBlog
export const updateCmsBlog = (data?: object) => {
   return http.request<resData>("post", "/cmsBlog/updateCmsBlog", { data });  
}

// @Tags CmsBlog
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsBlog/quickEdit", { data });  
}
 

// @Tags CmsBlog
// @Summary 用id查询CmsBlog
export const findCmsBlog = (params?: object) => {
   return http.request<resData>("get", "/cmsBlog/findCmsBlog", { params });  
}
 

// @Tags CmsBlog
// @Summary 分页获取CmsBlog列表
export const getCmsBlogList = (params?: object) => {
   return http.request<resData>("get", "/cmsBlog/getCmsBlogList", { params });  
}
 

// @Tags CmsBlog
// @Summary 导出excelCmsBlog列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsBlog/excelList", { params });  
}
